src/Components/utils/DarkMode.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { createContext, useState, useContext, useEffect, ReactNode } from 'react';
+
+interface DarkModeContextType {
+  darkMode: boolean;
+  toggleDarkMode: () => void;
+}
+
+const DarkModeContext = createContext<DarkModeContextType | undefined>(undefined);
+
+export const DarkModeProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const [darkMode, setDarkMode] = useState<boolean>(false);
+
+  useEffect(() => {
+    // Check if the user prefers dark mode using CSS media query
+    const prefersDarkMode = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Check with UX if users need to refresh the page in order for mode change to happen

Reply:
Yeah I think user do need to refresh the page for mode change

Reply:
Listener added, should be fixed

Reply:
It seems like I still have to refresh in order to change theme. But do check with UX to see which behavior they want.

Reply:
UX said they want it to be automatic, it's still working on my side. if you don't mind let's try to figure out the bug together tomorrow